atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,483 @@
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+        assertThat(listAllPhysicalTableNames(NON_DEFAULT_TABLE_PREFIX, NON_DEFAULT_OVERFLOW_TABLE_PREFIX))
+                .hasSameElementsAs(allTablesWithNonDefaultPrefix);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDoesNotDropTablesFromUnmappedTablesForMappedUser() {
+        namespaceDeleter.deleteAllDataFromNamespace();
+        createTableAndOverflowTableWithDefaultPrefixNoMapping(TABLE_NAME_ONE);
+        AllTableDetailsSnapshot snapshot = getTableDetailsForDefaultPrefixes();
+
+        createTableAndOverflowTable(TABLE_NAME_ONE);
+        namespaceDeleter.deleteAllDataFromNamespace();
+        assertThat(getTableDetailsForDefaultPrefixes()).isEqualTo(snapshot);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDoesNotDropTablesFromMappedTablesForUnmappedUser() {
+        namespaceDeleterWithDefaultPrefixNoMapping.deleteAllDataFromNamespace();
+        createTwoLogicalTablesWithDefaultPrefix();
+        AllTableDetailsSnapshot snapshot = getTableDetailsForDefaultPrefixes();
+
+        createTableAndOverflowTableWithDefaultPrefixNoMapping(TABLE_NAME_ONE);
+        namespaceDeleterWithDefaultPrefixNoMapping.deleteAllDataFromNamespace();
+        assertThat(getTableDetailsForDefaultPrefixes()).isEqualTo(snapshot);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDoesNotDropTimestampTable() throws SQLException {
+        new LegacyPhysicalBoundStoreStrategy(AtlasDbConstants.TIMESTAMP_TABLE, oracleDdlConfig.tablePrefix())
+                .createTimestampTable(connectionManager.getConnection(), _conn -> DBType.ORACLE);
+
+        createTableAndOverflowTableWithDefaultPrefixNoMapping(TABLE_NAME_ONE);
+        createTableAndOverflowTable(TABLE_NAME_ONE);
+        namespaceDeleter.deleteAllDataFromNamespace();
+        namespaceDeleterWithDefaultPrefixNoMapping.deleteAllDataFromNamespace();
+
+        assertSnapshotIsEmptyExcludingTimestampTable(getTableDetailsForDefaultPrefixes());
+    }
+
+    @Test
+    public void isNamespaceDeletedSuccessfullyReturnsFalseIfTablesRemain() {
+        createTableAndOverflowTable(TABLE_NAME_ONE);
+        assertThat(namespaceDeleter.isNamespaceDeletedSuccessfully()).isFalse();
+    }
+
+    @Test
+    public void isNamespaceDeletedSuccessfullyReturnsTrueIfNoTablesRemain() {
+        createTwoLogicalTablesWithDefaultPrefix();
+
+        namespaceDeleter.deleteAllDataFromNamespace();
+        assertThat(namespaceDeleter.isNamespaceDeletedSuccessfully()).isTrue();
+    }
+
+    @Test
+    public void isNamespaceDeletedSuccessfullyReturnsTrueEvenIfOtherPrefixExists() {
+        namespaceDeleter.deleteAllDataFromNamespace();
+        assertThat(namespaceDeleter.isNamespaceDeletedSuccessfully()).isTrue();
+        createTableAndOverflowTableWithNonDefaultPrefix(TABLE_NAME_ONE);
+
+        assertThat(namespaceDeleter.isNamespaceDeletedSuccessfully()).isTrue();
```

Comment:
nit: This is probably clearer if we just assert the non-default prefix?

Reply:
Ah. Fair enough, reading this comment with hindsight I'm not sure what it was referring to